File: js/containsDuplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const numberSet = new Set();

    for(let i = 0; i < nums.length; i++) {
        if (numberSet.has(nums[i])) {
            return true
        }

        numberSet.add(nums[i])
    }

    return false;
};

File: java/containsDuplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashMap<Integer, Integer> occurences = new HashMap();

        for (int i = 0; i < nums.length; i++) {
            if (occurences.containsKey(nums[i])) {
                return true;
            };

            occurences.put(nums[i], 1);
        }

        return false;        
    }
}